src/js/media-controller.js

```diff
@@ -171,6 +174,25 @@ class MediaController extends MediaContainer {
           document.exitPictureInPicture();
         }
       },
+      MEDIA_ENTER_CAST_REQUEST: () => {
+        const media = this.media;
+
+        if (!CastableVideo?.castEnabled) return;
+
+        // Exit fullscreen if needed
+        if (document[fullscreenApi.element]) {
+          document[fullscreenApi.exit]();
+        }
+
+        // Open the browser cast menu.
+        // Note this relies on a customized video[is=castable-video] element.
+        media.requestCast();
+      },
+      MEDIA_EXIT_CAST_REQUEST: async () => {
+        if (CastableVideo.castElement) {
+          CastableVideo.exitCast();
```

Comment:
potential blocker: confirm this won't have issues with multiple `CastableVideo`s on the same page.

Reply:
you're right here I think but we're doing this for pip, fullscreen, etc all like this.
it's fairly safe because these exit requests are only made from a button that is in the exit state.
we should revisit it for all but this will be the same difficulty with the nested shadow dom etc.
prefer to wait a little because it's unlikely to reproduce this bug

Reply:
ok I'm cool with treating this as a fast follow 👌. Let's make sure we get an issue filed for this.